### PR TITLE
Reveal in finder in search 2

### PR DESCRIPTION
**Pull request by pskowronek**
Fix addressing bug https://github.com/mucommander/mucommander/issues/591 to Reveal In XYZ work when:

    in search results
    clicked on '..'

It replaces https://github.com/mucommander/mucommander/pull/628

### CODE REVIEW

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/RevealInDesktopAction.java [hunk 1]
```diff
@@ -55,21 +56,25 @@ public RevealInDesktopAction(MainFrame mainFrame, Map<String, Object> properties
...
     protected void toggleEnabledState() {
         AbstractFile currentFolder = mainFrame.getActivePanel().getCurrentFolder();
+        AbstractFile selectedFile = mainFrame.getActiveTable().getSelectedFile();
         setEnabled(currentFolder.getURL().getScheme().equals(LocalFile.SCHEMA)
                 && !currentFolder.isArchive()
-                && !currentFolder.hasAncestor(AbstractArchiveEntryFile.class));
+                && !currentFolder.hasAncestor(AbstractArchiveEntryFile.class)
+                || currentFolder.getURL().getScheme().equals(SearchFile.SCHEMA)
+                && selectedFile != null
+                && selectedFile.getURL().getScheme().equals(LocalFile.SCHEMA));
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/action/impl/RevealInDesktopAction.java:65]: we'd still want to disable it for archive entries that appear in the search results, right?

Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/RevealInDesktopAction.java:65]: I've checked that and in archives 'reveal in XYZ' doesn't appear. The condition that parent is `Search` schema and the file must be `LocalFile` should be enough. Unless I don't see sth.

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/ParentFolderAction.java [hunk 2]
```diff
@@ -65,10 +65,7 @@ public ParentFolderAction(MainFrame mainFrame, Map<String, Object> properties) {
 
     private void toggleEnabledStateAdapter() {
         AbstractFile currentFolder = mainFrame.getActivePanel().getCurrentFolder();
-        if (currentFolder.getURL().getScheme().equals(SearchFile.SCHEMA))
-            setEnabled(false);
-        else
-            toggleEnabledState();
+        toggleEnabledState();
```
[the remark this answers is not in the document]
Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/ParentFolderAction.java:68]: I will fix `RevealInDesktopAction` to use `ActiveTabAction` and revert changes in `ParentFolderAction` (though I will put braces to if/else over there).

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/RevealInDesktopAction.java (hunk 1 above)
Comment (ahadas) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/RevealInDesktopAction.java:65]: right

**Review by ahadas — APPROVED** [review]
thanks!

**Pull request merged**